contracts/GSN/IRelayRecipient.sol

```diff
@@ -1,14 +1,32 @@
 pragma solidity ^0.5.0;
 
-/*
- * @dev Interface for a contract that will be called via the GSN from RelayHub.
+/**
+ * @dev Base interface for a contract that will be called via the GSN from {IRelayHub}. Do not use directly, inherit from
+ * {GSNRecipient} instead.
  */
 contract IRelayRecipient {
     /**
-     * @dev Returns the address of the RelayHub instance this recipient interacts with.
+     * @dev Returns the address of the {IRelayHub} instance this recipient interacts with.
      */
     function getHubAddr() public view returns (address);
 
+    /**
+     * @dev Called by {IRelayHub} to validate if this recipient accepts being charged for a relayed call. Note that the
+     * recipient will be charged regardless of the execution result of the relayed call (i.e. if it reverts or not).
+     *
+     * The relay request was originated by `from` and will be served by `relay`. `encodedFunction` is the relayed call
+     * calldata, so its first four bytes are the function selector. The relayed call will be forwarded `gasLimit` gas,
+     * and the transaction executed with a gas price of at least `gasPrice`. `relay`'s fee is `transactionFee`, and the
+     * recipient will be charged at most `maxPossibleCharged` (in wei). `nonce` is the sender's (`from`) nonce for
+     * replay attack protection in {IRelayHub}, and `approvalData` is a optional field that can be used to hold a signature
+     * over all or some of the previous values.
+     *
+     * Retuns a tuple, where the first value is used to indicate approval (0) or rejection (custom non-zero error code,
```

Comment:
Should we use `@return` tag https://solidity.readthedocs.io/en/v0.5.11/natspec-format.html#tags ?

Fix typo:
```suggestion
     * Returns a tuple, where the first value is used to indicate approval (0) or rejection (custom non-zero error code,
```